clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -799,6 +803,9 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 // treat subscribing to empty topic list as the same as unsubscribing
                 this.unsubscribe();
             } else {
+                if (this.autoCommitEnabled)
+                    // make sure the offsets of topic partitions the consumer is unsubscribing from are committed
```

Comment:
Maybe add "since these partitions will not be included in the revocation callback of the next rebalance"?